### ISSUE TITLE
Can I suppress "Expression value is unused" for my overridden operator?

### ISSUE THREAD

**Issue by ArcticLampyrid** [microsoft/pyright#5264]
Hi, I've overridden the operator `__lshift__` and do something with side effect, then can I suppress "Expression value is unused" for the caller?

```python
def build(q1, q2):
    builder = MyBuilder()
    builder << "abc" # "Expression value is unused" is reported
    builder << "def" # "Expression value is unused" is reported
    return builder.getvalue()

class MyBuilder:
    def __init__(self):
        self.result = StringIO()
    def __lshift__(self, other):
        # just as you see, it's with side effect
        self.result.write(other)
        self.result.write('\n')
        return self
    def getvalue(self):
        return self.result.getvalue()[0:-1]
```

**Comment by erictraut** [comment]
Yes, you can disable the `reportUnusedExpression` diagnostic check for your entire project (by specifying so in the config file or your LSP settings), for a single file (by adding a `# pyright: reportUnusedExpression=false` comment to the top of the file) or on a per-line basis (by adding a `#pyright: ignore[reportUnusedExpression]` comment to the end of the line).

That said, my recommendation is to not write a `__lshift__` method that has a side effect. Doing so will make your code less understandable and will break assumptions made by tools and other developers alike. If you really want to use the `<<` operator, the `<<=` (`__ilshift__`) would be more appropriate for an operation that has side effects. Or if that doesn't work, then I'd recommend using a method with an appropriate name rather than a built-in operator.

**Comment by ArcticLampyrid** [comment]
Well, I'm wondering that can I just disable this check on specific class but not globally?

**Comment by joamatab** [comment]
yes, ideally we can define a pyright ignore in pyproject.toml


@nikosavola

**Comment by Throne3d** [comment]
Hi there! I was wondering, did pyright ever add a configuration to disable this warning on specific classes' operators? I considered opening a new issue but it seems very related to this one so I thought it best to post here first!

With [Apache Airflow](https://github.com/apache/airflow), they override the lshift + rshift operators to define dependencies between tasks in a DAG, and I see a lot of code that triggers this "expression value is unused" warning as a result. So far I've been suppressing this warning in several projects to reduce warning fatigue here, but it either means a lot of `type: ignore` / `pyright: ignore[reportUnusedExpression]` over these (several) lines, or disabling the diagnostic for whole files/projects, which means we lose the benefit of the diagnostic catching more subtle bugs.

All operators + sensors in Airflow eventually inherit from [a DependencyMixin class with the overrides](https://github.com/apache/airflow/blob/2.7.3/airflow/models/taskmixin.py#L96). The code can be rewritten to use `.set_downstream()` or `.set_upstream()` instead, but I rarely see code use that, and it can make chained dependencies a lot longer. Example code might look like:

```py
import datetime
from airflow import DAG
from airflow.operators.empty import EmptyOperator
from airflow.sensors.time_delta import TimeDeltaSensorAsync

with DAG(dag_id="sample-dag") as dag:
    wait = TimeDeltaSensorAsync(task_id="wait", delta=datetime.timedelta(seconds=30))
    finish = EmptyOperator(task_id="finish")

    wait >> finish  # Expression value is unused
```

For an example of chained and branched dependencies, where this rewrite can make these dependencies less clear:
```py
    task1 >> task2 >> task3 >> task4
    # becomes
    task1.set_downstream(task2)
    task2.set_downstream(task3)
    task3.set_downstream(task4)

    task1 >> [task2a, task2b] >> task3 >> task4
    # becomes
    task1.set_downstream([task2a, task2b])  # equivalent to task1.set_downstream(task2a) and task1.set_downstream(task2b)
    task3.set_upstream([task2a, task2b])
    task3.set_downstream(task4)
```

It would be great to have some way to disable the warning on classes inheriting from `airflow.models.taskmixin.DependencyMixin`'s `__lshift__` and `__rshift__` (and `__rrshift__` and `__rlshift__` for supporting interactions with lists of operators/sensors).

**Comment by Throne3d** [comment]
Submitted a new issue in https://github.com/microsoft/pyright/issues/6612 =)
